library/network_connections.py

```diff
@@ -960,6 +961,17 @@ def connection_create(self, connections, idx, connection_current=None):
         else:
             raise MyError("unsupported type %s" % (connection["type"]))
 
+        if connection["type"] == "wireless":
+            s_wireless = self.connection_ensure_setting(con, NM.SettingWireless)
+            s_wireless.set_property(
+                NM.SETTING_WIRELESS_CLONED_MAC_ADDRESS, connection["cloned_mac"]
+            )
+        else:
+            s_wired = self.connection_ensure_setting(con, NM.SettingWired)
+            s_wired.set_property(
+                NM.SETTING_WIRED_CLONED_MAC_ADDRESS, connection["cloned_mac"]
+            )
```

Comment:
if `connection["cloned_mac"] is None`, then `s_wired.set_property()` indeed has no effect. However, we would still call `self.connection_ensure_setting()` and add a setting. That seems not desirable. For example, if you add a bridge profile, then adding an otherwise empty `[ethernet]` setting seems wrong. Let's add this block inside a
```
    if connection["cloned_mac"]:
        if connection["type"] == "wireless":
            ...
```

Reply:
True, I did the change